### PR TITLE
Add Opcodes details and parametrization required for measurements

### LINKED ISSUE #57: Compile a detailed rundown of the specifics of each OPCODE

### PR DESCRIPTION
**Pull request by pdobacz**
Closes #57

### CODE REVIEW

File: docs/opcodes_details.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+0x1b,SHL,3,verylow,2,1,Left shift operation,
+0x1c,SHR,3,verylow,2,1,Logical right shift operation,
+0x1d,SAR,3,verylow,2,1,Arithmetic (signed) right shift operation,
+```
+
+They pop and push, or pop and peek and then modify stack in-place.
+
+### Parameters
+
+2 or 3 values on the stack, should not have impact on cost.
```
Comment (chfast) [inline docs/opcodes_details.md:67]: Many of these instruction are actually sensitive to the input arguments. One reason is that argument values may be out of the core instruction operational range in such case the output being some arbitrary values (e.g. left shift by large value produces 0 but this case may be handled separately in the implementation). Other reason is implementation complexity which may have multiple control-flow paths to follow (e.g. my division implementation actually has 3 different algorithms depending on the divisor size).

File: docs/opcodes_details.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+`0x90 -- 0x9f,SWAP*,3,verylow,* + 1,* + 1,Exchange 1st and (* + 1)th stack items.,`
+
+straightforward in all impls. No comments.
+
+## `0xf3,RETURN` and `0xfd,REVERT`
+
+`0xf3,RETURN,0,zero,2,0,Halt execution returning output data.,`
+`0xfd,REVERT,,,2,0,End execution, revert state changes, return data mem[p…(p+s)),`
+
+- **evmone** - does a check_memory (**TODO** clarify with PB) and also does a micro-optimization for `size == 0`
```
Comment (chfast) [inline docs/opcodes_details.md:322]: The `RETURN` / `REVERT` works similar to other instructions that copy from/to memory so it needs memory bounds check and possibly charge more gas if case such memory size is not allocated yet.

The "micro-optimization" comes from a pedantic static analysis / instrumentation tools such that you are not allowed to produce invalid pointer even though you will never access it. But `RETURN` of size zero may be measured separately.

File: docs/opcodes_details.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+`0x5a,GAS,2,base,0,1,"Get the amount of available gas, including the corresponding reduction for the cost of this instruction.",`
+
+Implementation differ due to different approaches to gas tracking, but I don't see any hidden parameters to account for.
+
+## `0x5b,JUMPDEST`
+
+`0x5b,JUMPDEST,1,,0,0,Mark a valid destination for jumps,`
+
+- **geth** noop
+- **evmone** not a noop, flushes some gas calculations for the code block. Note - JUMPDEST is rewriten to a `evmone`-specific OPX_BEGINBLOCK
```
Comment (chfast) [inline docs/opcodes_details.md:291]: This is true for evmone/Advanced where gas cost and stack checks are done per instruction basic blocks. However, newer evmone version also has Baseline interpreter. If you are interested in checking the new one too I can rebase your instrumentation.

**Review by chfast — APPROVED** [review]

**Review by pdobacz — COMMENTED** [review]
thanks @chfast, doc updated

File: docs/opcodes_details.md [hunk 4]
```diff
@@ -0,0 +1,345 @@
+# Rundown of OPCODE details
+
+This file contains a detailed rundown of all EVM OPCODEs concerned, with notes on their specifics and what to watch our for.
+
+Based on `src/program_generator/data/opcodes.csv` and `src/program_generator/data/selection.csv`.
+
+## Takeaways
+
+See below for a detailed OPCODE-by-OPCODE description of parameters and concerns.
+This is a proposed list of things to tackle, in order from important to negligible:
```
Comment (JacekGlen) [inline docs/opcodes_details.md:10]:
```suggestion
This is a proposed list of things to tackle, in order of importance:
```

**Pull request merged**